app/Http/Controllers/api/StudentProjectsDetailController.php

```diff
@@ -1,35 +1,39 @@
 <?php
 
+declare(strict_types=1);
+
 namespace App\Http\Controllers\api;
 
 use App\Http\Controllers\Controller;
-use App\Models\Student;
-use App\Models\Project;
+use App\Service\Resume\StudentProjectsDetailService;
+use Illuminate\Http\JsonResponse;
+use App\Exceptions\StudentNotFoundException;
+use App\Exceptions\ResumeNotFoundException;
+use App\Exceptions\ProjectNotFoundException;
+
 
 class StudentProjectsDetailController extends Controller
 {
-    public function __invoke($uuid)
-    {
-        $student = Student::where('id', $uuid)->with('resume')->firstOrFail();
-        $resume = $student->resume;
-
-        $projectIds = json_decode($resume->project_ids);
+    private $studentProjectsDetailService;
 
-        $projects = Project::findMany($projectIds);
-
-        $projects_detail = [
-            'projects' => $projects->map(function ($project) {
-                return [
-                    'uuid' => $project->id,
-                    'project_name' => $project->name,
-                    'company_name' => $project->company->name,
-                    'project_url' => $project->project_url,
-                    'tags' => json_decode($project->tags),
-                    'github_url' => $project->github_url,
-                ];
-            })
-        ];
+    public function __construct(StudentProjectsDetailService $studentProjectsDetailService)
+    {
+        $this->studentProjectsDetailService = $studentProjectsDetailService;
+    }
 
-        return response()->json($projects_detail);
+    public function __invoke($uuid): JsonResponse
+    {
+        try {
+            $service = $this->studentProjectsDetailService->execute($uuid);
+            return response()->json(['projects' => $service]);
+        } catch (StudentNotFoundException $e) {
+            return response()->json(['message' => $e->getMessage()], 404);
```

Comment:
According to the previous comment I've made in ProjectNotFoundException file, the hard-coded 404 code could be replaced with ``$e->getCode()``.

And so with the rest of the exceptions.

Reply:
Thanks 
:+1:
:thumbsup: